geochem_inverse_optimize.py

```diff
@@ -149,21 +169,43 @@ def _build_primary_terms(self) -> None:
 
     def _build_regularizer_terms(self) -> None:
         # Build the regularizer
-        for adjacent_nodes, border_length in self.sample_adjacency.items():
-            a_concen = self.sample_network.nodes[adjacent_nodes[0]]["data"].my_value
-            b_concen = self.sample_network.nodes[adjacent_nodes[1]]["data"].my_value
-            # TODO: Make difference a log-ratio
-            # self._regularizer_terms.append(border_length * (cp_log_ratio(a_concen,b_concen)))
-            # Simple difference (not desirable)
-            self._regularizer_terms.append(border_length * (a_concen - b_concen))
+        if self.continuous:
```

Comment:
Since there's no real common code here, could you please make two new functions `_build_regularizer_terms_continuous` and `_build_regularizer_terms_TODO` (replace `TODO` with whatever you're calling the non-continuous thing) and have this function call whichever is appropriate based on `self.continuous`?